### ISSUE TITLE
Add italian language

### ISSUE THREAD

**Issue by sommoomnia** [MikeBland/mbtx#3]
Add italian language into the firmware please!

**Comment by MikeBland** [comment]
Is this for er9x or ersky9x?
I'm happy to add other languages, although doing so on er9x causes me problems as I would need to build several new .hex files for any new language.
ersky9x is easier as all the alnguages are 'built in'.
In either case, I need assistance with the translation. The best for me is if someone takes one of the existing language files (e.g. en.h, de.h, fr.h . . ) and translates the text to give, in this case, an it.h file.

**Comment by sommoomnia** [comment]
ersky9x I have a turnigy 9xrpro... I can tale this file?

**Comment by MikeBland** [comment]
Yes, take any one of the files and do the translation.
Note that many lists are like:
#define ISTR_MET_IMP        "\003MetImp"
where the \003 indicates each item is 3 characters long "Met" and "Imp" (for Metric or Imperial).
while some simple things like:
#define ISTR_15_ON          "\015On"
Where the \015 indicates "move to 13 characters across the display". The \015 is in octal (base 8) so is 0*64 +1*8 + 5 = 13.

**Comment by L4ky** [comment]
Hi, i don't have enuogh time to study how to recompile etc.
If you give me the language file i could translate it in Italian and i will send it back to you, so you can recompile the Italian version of this firmware. 
Maybe you don't believe but in Italy many people uses this firmware on theirs radios!

**Comment by sommoomnia** [comment]
https://github.com/MikeBland/mbtx/blob/master/radio/ersky9x/src/en.h

here is the English files of the developer and I attach the file that I
started translating I ... if someone wants to perfect it!
https://drive.google.com/open?id=0Bw_PvZENe5ELcDhIZ1JyaVhvV1E

2015-09-28 17:54 GMT+02:00 L4ky <notifications@github.com>:

**Comment by sommoomnia** [comment]
https://docs.google.com/document/d/1PJIBylAQ3UOShYtVJVlTbtZtw68e3ylXyRRDe2AfBV8/edit?usp=sharing

I just noticed that the file I uploaded not be changed ... if you want you
can directly edit this doc .... then I create .h

2015-09-28 17:58 GMT+02:00 francesco Piccininni <sommoomnia@gmail.com>:

**Comment by L4ky** [comment]
I don't know if it's better translate some strings or not, because many strings in italian are much longer than english. I think we risk that some strings can't be displayed cause they're too long.

Example:
English: Press any key to skip
Italian: Premi qualsiasi tasto per saltare

As you can see the italian translation is much longer... how do we act in this situations?

**Comment by sommoomnia** [comment]
True story ;)
Il 29/set/2015 10:39 AM, "L4ky" <notifications@github.com> ha scritto:

**Comment by L4ky** [comment]
I translated some mores strings. Some of them i prefered to leave them in english since there is no effective translation in Italian and it sounds better.

**Comment by MikeBland** [comment]
I've added the code to support Italian, so it will appear in the list of languages. I'll post a test version soon (here: http://openrcforums.com/forum/viewtopic.php?f=7&t=4676#p65894)

There is a problem with some translations, e.g.:
#define ISTR_KEYNAMES       "\005 Menu Exit Down   UpRight Left"
to
#define ISTR_KEYNAMES       "\005 Menu Esci Giu  Su Destra Sinistra"
This is an example of a selection list. The \005 indicates there are exactly 5 characters in each item, and the menus are usually designed with this known, so making the names longer may need the menus adjusting.
Is it possible to abbreviate Destra and Sinistra, so they are only 5 characters long?

**Comment by L4ky** [comment]
Of course, DX or Des = Destra; SX or Sin = Sinistra

**Comment by MikeBland** [comment]
OK, I'll use Des. and Sin.
Next, you have:
#define ISTR_SHIFT_SEL      " Polarità"
the à will need to be treated as a "special character, and I'll need to create the correct "extraFont" file to handle it.
e.g. French uses:
// Special characters:
// é  use \300
// è  use \301
// à  use \302
// î  use \303
// ç  use \304

So, in the fr.h file the above line would be:
#define ISTR_SHIFT_SEL      " Polarit\302"

What other special characters do you need?

**Comment by MikeBland** [comment]
I've posted a test version of ersky9x and ersky9xr on the thread I linked to above. This includes support for the Italian language.

**Comment by L4ky** [comment]
I have a Turnigy 9X flashed with er9x. Can i flash ersky9x firmware or it will brick?
In my turnigy there is a ATMEGA128 ( not 64 )

Sorry i'm a newbie in this world, i flashed the 9x for the first time 2 days ago..

**Comment by MikeBland** [comment]
No, ersky9x is for boards using an ARM processor (SKY board, 9XR-PRO, Taranis, 9Xtreme).
What you will want to do, if you haven't already, is use er9x-128.hex. This is specific for the '128 processor and includes some extra features that don't fit in the '64.

**Comment by clapzen** [comment]
Add Vietnamese language into the firmware please!
you have sample code english (en.h) can be shared. I can translate into Vietnamese language (vi.h)
thanks!

**Comment by DaniSpringer** [comment]
Do you need Italian translation? I can help.
